lib/iris/tests/_shared_utils.py

```diff
@@ -207,17 +206,36 @@ def get_result_path(relative_path):
     return os.path.abspath(os.path.join(_RESULT_PATH, relative_path))
 
 
-def result_path(basename=None, ext=""):
+def _check_for_request_fixture(request, func_name: str):
+    """Raise an error if the first argument is not a pytest.FixtureRequest.
+
+    Written to provide the clearest possible message for devs refactoring from
+    the deprecated IrisTest style tests.
+    """
+    if not hasattr(request, "fixturenames"):
+        message = (
+            f"{func_name}() expected: pytest.FixtureRequest instance, got: "
+            f"{request}"
+        )
+        raise ValueError(message)
+
+
+def result_path(request: pytest.FixtureRequest, basename=None, ext=""):
     """Generate the path to a test result; from the calling file, class, method.
 
     Parameters
     ----------
+    request : pytest.FixtureRequest
```

Comment:
Perhaps make it clear that all it needs to have is the word request, and pytest does the rest?

Reply:
6829c759d9017499dca3fd2bc185a2aaa83a8e02